### PR TITLE
Expand documentation for the main structures.

### PR DESCRIPTION
**Pull request by afck**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/872)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @brian-js (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: src/authority.rs [hunk 1]
```diff
@@ -19,10 +19,12 @@ use xor_name::XorName;
 use sodiumoxide::crypto::{hash, sign};
 use std::fmt::{Debug, Formatter};
 
-/// Persona types recognised by network.
+/// An entity that can act as a source or destination of a message.
+///
+/// An `Authority` can be an individual client or node, or a group of nodes.
```
Comment (dirvine) [inline src/authority.rs]: This maybe should be more explicit. So node, NodeManager, NaeManager, ClientManager

File: src/client.rs [hunk 2]
```diff
@@ -40,10 +43,12 @@ pub struct Client {
...
 
 impl Client {
-    /// Starts a new RoutingIdentity, which will also start a new Core.
-    /// The Core will only bootstrap to the network and not attempt to
-    /// achieve full routing node status.
-    /// If the client is started with a relocated id (ie the name has been reassigned),
+    /// Create a new `Client`.
+    ///
+    /// It will automatically connect to the network, but not attempt to achieve full routing node
+    /// status. The name of the client will be the name of the `PublicId` of the `keys`.
```
Comment (dirvine) [inline src/client.rs]: and cryptographically secure, where groups use group consensus

File: src/immutable_data.rs [hunk 3]
```diff
@@ -22,18 +22,20 @@ use xor_name::XorName;
 use sodiumoxide::crypto;
 
 #[derive(Hash, Clone, Eq, PartialEq, Ord, PartialOrd, RustcEncodable, RustcDecodable, Debug)]
-/// ImmutableDataType.
+/// The type of an individual copy of immutable data.
```
Comment (dirvine) [inline src/immutable_data.rs]: Self validating data - i.e. name = Hash(content) or for backup name = Hash(Hash(content) etc.

File: src/plain_data.rs [hunk 4]
```diff
@@ -18,7 +18,7 @@
 use rustc_serialize::{Decoder, Encodable, Encoder};
 use xor_name::XorName;
 
-/// PlainData
+/// Plain data with a name and a value.
```
Comment (dirvine) [inline src/plain_data.rs]: Name is set and not derived in this case. Allowing for contents to be set by user i.e. may contain and be restricted to a DSL

File: src/structured_data.rs [hunk 5]
```diff
@@ -62,8 +65,8 @@ impl StructuredData {
         Ok(structured_data)
     }
 
-    /// This is a static function required for computing a name give the tag-type and identifier to
```
Comment (dirvine) [inline src/structured_data.rs]: The calculation is set by routing,

File: src/immutable_data.rs (hunk 3 above)
Comment (afck) [inline src/immutable_data.rs]: Done.
(Added it to the documentation of the enum variants and of `ImmutableData`.)

File: src/structured_data.rs (hunk 5 above)
Comment (afck) [inline src/structured_data.rs]: I'm not entirely sure what you mean. I expanded the text to make it more precise; is the new version correct?

Comment (dirvine) [inline src/structured_data.rs]: :+1:

File: src/plain_data.rs (hunk 4 above)
Comment (afck) [inline src/plain_data.rs]: Do you mean a Domain-Specific Language?
I added that any restrictions are left to the implementation - or should I explicitly mention DSLs?

Comment (dirvine) [inline src/plain_data.rs]: Yes I did, but it is fine Andreas, much better shape now. Thx for these PR's they have helped a lot.

**Pull request merged**